src/pipecat/services/anthropic.py

```diff
@@ -284,20 +284,16 @@ async def _process_context(self, context: OpenAILLMContext):
                 cache_read_input_tokens=cache_read_input_tokens,
             )
 
-    async def _update_settings(self, frame: LLMUpdateSettingsFrame):
-        if frame.model is not None:
-            logger.debug(f"Switching LLM model to: [{frame.model}]")
-            self.set_model_name(frame.model)
-        if frame.max_tokens is not None:
-            await self.set_max_tokens(frame.max_tokens)
-        if frame.temperature is not None:
-            await self.set_temperature(frame.temperature)
-        if frame.top_k is not None:
-            await self.set_top_k(frame.top_k)
-        if frame.top_p is not None:
-            await self.set_top_p(frame.top_p)
-        if frame.extra:
-            await self.set_extra(frame.extra)
+    async def _update_settings(self, settings: Dict[str, Any]):
+        for key, value in settings.items():
+            setter = getattr(self, f"set_{key}", None)
+            if setter and callable(setter):
```

Comment:
I'm wondering if it's worth it to have a `self._settings: Dict[str, Any]` in the class instead if one field per setting.

Then you could just do:

```
for key, value in settings.items():
  if key in settings:
      self._settings[key] = value
  else:
      logger.warning(f"Unknown setting for Anthropic LLM service: {key}")
```

and in the constructor we initialize `self._settings = { "key": value, .... }`

Reply:
It's just that

```
            setter = getattr(self, f"set_{key}", None)
            if setter and callable(setter):
```

feels a bit wrong.